### PR TITLE
New Docs Site

### PR DESCRIPTION
**Pull request by gojutin**
This PR is primarily responsible for introducing a new VitePress docs site (work in progress) located under the `website` directory. This new site includes a revamped documentation structure, updated docs, and embedded Typedoc API docs.

The latest version of the site is available here for viewing: https://twind-dev.netlify.app/

Other changes include:
- Updated Typedoc configuration
- Removed husky hook temporarily due to inability to commit
- Removed  `doctoc` commands and dependency
- Removed original docs and module READMEs
- Simplified main root README

### CODE REVIEW

**Comment by sastan** [comment]
Awesome job!!! This is really something. I'll try to put in some time this weekend to get a initial version out as fast as possible. We can then regroup and improve from there.

**Comment by gojutin** [comment]
Thanks!  I still plan to do a lot more to update docs and  SEO (better descriptions in frontmatter, favicon, etc.) But, I'll give it a break for the weekend and give you a chance to get the initial version launched. Once we have it live, we can sign up for [Algolia DocSearch](https://docsearch.algolia.com/) to get the search bar working. Should be as simple as signing up and supplying an API key/index.

**Comment by gojutin** [comment]
If there is anything I can do to make this PR easier to merge, please just let me know.

**Comment by gojutin** [comment]
Remaining work to be done:

- [ ] Review/complete initial version of the docs
- [ ] Determine the best order of docs and reorder accordingly
- [ ] Add email address to `handbook/code-of-conduct.md`
- [ ] Register for Algolia DocSearch and supply API key/index to `.vitepress/config.js` as env var
- [ ] Replace all Typedoc link syntax with internal/external links
- [ ] Add Favicon, alt images
- [ ] Update docs frontmatter descriptions for improved SEO
- [ ] Review head for improvements to SEO
- [ ] Review/update all badges based on changes
- [ ] Sort out CI/CD process for docs (spellcheck, automatic Typedoc builds, etc.)

**Comment by sastan** [comment]
Thanks again! I'm currently trying to use the typescript markdown plugin to generate the API docs which would then be used by vitepress. I'll let you know what I come up with.

**Comment by sastan** [comment]
@gojutin I've made some minor changes:

- renamed website/ to docs/
- use typedoc markdown as input for vitepress
- re-based on the current main branch
- add husky back – if you have any issues please tell me

**Comment by sastan** [comment]
@gojutin Totally forget. You have put in some gems like the new root README, the landing page with the advantages… Great stuff. Thank you for your time on this.

**Comment by gojutin** [comment]
No problem! Glad to hear you are liking the changes.

File: docs/handbook/the-shim.md [hunk 1]
```diff
@@ -73,6 +73,77 @@ Internally `twind/shim` uses `twind/observe`, which may be useful for advanced u
...
 ```
 
+The `twind/shim` module utilizes the {@link twind/observe} module internally, but it provides its own {@link setup} function for customizing the used {@link tw} instance and setting the target node to be shimmed. It also provides a {@link disconnect} function to stop shimming/observing all nodes.
+
+```js
+import { setup, disconnect } from 'twind/shim'
+
+setup({
+  // node element to shim/observe (default: document.documentElement)
+  target: document.querySelector('#__twind'),
```
Comment (danielweck) [inline docs/handbook/the-shim.md:83]: Isn't the `target` property supposed to reference the DOM fragment to observe? (typically the `document.documentElement` or `document.body` element).
In this example, the `#__twind` CSS selector references Twind's own `<style id="__twind"></style>`.

Comment (sastan) [reply docs/handbook/the-shim.md:83]: 😄 You are totally right! Thank you for the report.

Comment (sastan) [inline docs/handbook/the-shim.md:83]:
```suggestion
  target: document.querySelector('main'),
```

**Comment by danestves** [comment]
I just want to pass here to tell you thank you for this docs https://twind-dev.netlify.app/usage-guides/nextjs.html it will be helpful to me to integrate with NextJS

**Comment by sastan** [comment]
> I just want to pass here to tell you thank you for this docs https://twind-dev.netlify.app/usage-guides/nextjs.html it will be helpful to me to integrate with NextJS

@danestves Please use [@twind/next](https://github.com/tw-in-js/use-twind-with/tree/main/packages/next). We are updating the docs to use this new package.

**Comment by danestves** [comment]
> > I just want to pass here to tell you thank you for this docs https://twind-dev.netlify.app/usage-guides/nextjs.html it will be helpful to me to integrate with NextJS
> 
> @danestves Please use [@twind/next](https://github.com/tw-in-js/use-twind-with/tree/main/packages/next). We are updating the docs to use this new package.

I will use it when you launch [#148](https://github.com/tw-in-js/twind/pull/148) 😉

**Pull request closed without merging**